runtime/compiler/optimizer/J9EstimateCodeSize.cpp

```diff
@@ -1599,20 +1602,23 @@ TR_J9EstimateCodeSize::realEstimateCodeSize(TR_CallTarget *calltarget, TR_CallSt
       }
 
    if (isCandidate)
-      _optimisticSize += calltarget->_partialSize;
+      _analyzedSize += calltarget->_partialSize;
    else
-      _optimisticSize += calltarget->_fullSize;
+      _analyzedSize += calltarget->_fullSize;
 
    int32_t sizeThreshold = _sizeThreshold;
    if (isCandidate)
       sizeThreshold = std::max(4096, sizeThreshold);
-   ///if(_optimisticSize > _sizeThreshold)   // even optimistically we've blown our budget
-   heuristicTrace(tracer(),"--- Depth %d: Checking Optimistic size vs Size Threshold: _optimisticSize %d _sizeThreshold %d sizeThreshold %d ",_recursionDepth, _optimisticSize, _sizeThreshold, sizeThreshold);
+   ///if(_analyzedSize > _sizeThreshold)   // even optimistically we've blown our budget
+   heuristicTrace(tracer(),"--- Depth %d: Checking Analyzed size vs Size Threshold: _analyzedSize %d _sizeThreshold %d sizeThreshold %d ",_recursionDepth, _analyzedSize, _sizeThreshold, sizeThreshold);
+
+   static const char *af = feGetEnv("TR_AnalyzedAllowanceFactor");
+   static const int32_t allowanceFactor = af ? atoi(af) : DEFAULT_ANALYZED_ALLOWANCE_FACTOR;
 
-   if (_optimisticSize > sizeThreshold) // even optimistically we've blown our budget
+   if (_analyzedSize > allowanceFactor*sizeThreshold) // even optimistically we've blown our budget
```

Comment:
Just for my understanding, it looks like we increasing the threshold here, not decreasing?

Reply:
The changes avoid resetting the `_analyzedSize` when we backtrack. So, in that sense the logic is not directly comparable, since that is a change in how we go about this estimate code size analysis at a high level. But given that we avoid resetting the `_analyzedSize` when we backtrack, it will contain a larger value as we estimate (analyze) the call graph than it did before. Before this change though, there would be no limit at all on the amount we might estimate (analyze) due to backtracking, but now, we have a limit that is 2x the original limit that applies to the amount that we estimate (analyze). The original limit still gets compared against the `_realSize` to still limit how much of the call graph we bring in to potentially inline.

Reply:
Thanks!